src/functions/bootstrap.php

```diff
@@ -16,3 +18,35 @@ function drupal_get_filename(string $type, string $name, ?string $filename = nul
     assert($pathResolver instanceof ExtensionPathResolver);
     return $pathResolver->getPathname($type, $name);
 }
+
+function drupal_get_schema(?string $table = null, ?bool $rebuild = false): array|bool
+{
+    $loader = require realpath(InstalledVersions::getRootPackage()['install_path']) . '/vendor/autoload.php';
+    $loader->add('Drupal\\TestTools', InstalledVersions::getInstallPath('drupal/drupal') . '/core/tests');
+    $schema = &drupal_static(__FUNCTION__);
+    if (!isset($schema) || $rebuild) {
+        if (!$rebuild && ($cached = \Drupal::cache()->get(__FUNCTION__))) {
+            $schema = $cached->data;
+        } else {
+            $schema = [];
+            $module_handler = \Drupal::moduleHandler();
+            foreach ($module_handler->getModuleList() as $name => $module) {
+                $schema += SchemaInspector::getTablesSpecification($module_handler, $name);
+            }
+            \Drupal::cache()->set(__FUNCTION__, $schema);
+        }
+    }
+    if (!isset($table)) {
+        return $schema;
+    }
+    if (isset($schema[$table])) {
+        return $schema[$table];
+    } else {
+        return false;
+    }
```

Comment:
Why does this need to add TestTools? In https://git.drupalcode.org/project/drupal/-/blob/7.x/includes/bootstrap.inc#L3400 it re-invokes hook_schema of modules. It doesn't actually inspect the current installed schema.

Reply:
I was trying to avoid duplicating the code in [SchemaInspector::getTablesSpecification()](https://api.drupal.org/api/drupal/core%21tests%21Drupal%21TestTools%21Extension%21SchemaInspector.php/function/SchemaInspector%3A%3AgetTablesSpecification/10). SchemaInspector doesn't actually inspect the currently installed schema. It was the suggested replacement for drupal_get_module_schema() in the Drupal [change record](https://www.drupal.org/node/2970993).

Reply:
Oh.. interesting! I should have researched the change record first. Hm. Interesting on which approach